src/globals.c

```diff
@@ -318,25 +310,48 @@ bool init_globals(void)
 		return false;
 	}
 
-/* Set configuration defaults based on options provided
-	at configure time.  */
+/* Set defaults with following order of preference:
+	1. Runtime flags
+	2. State dir configuration files
+	3. Configure time settings
+*/
+	if (!set_format_string(NULL)) {
```

Comment:
Using NULL to mean "use the default value" was a convention I adopted when I refactored the globals code a while back. But I think I only documented it in commit descriptions.